### PR TITLE
FOGL-6642 Improve documentation

### PR DESCRIPTION
**Pull request by MarkRiddoch**
Signed-off-by: Mark Riddoch <mark@dianomic.com>

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -155,14 +171,115 @@ Examining the content of the Python, a few things to note are;
...
+
+  - Fledge is designed to be managed solely via the Fledge API and applications that use that API. There is no facility within that API to manage arbitrary files within the filesystem.
+
+It is common to make use of files during development of a script to write information to in order to aid development and debugging, however this should be removed, along with associated imports of packages required to perform the file IO, when a filter is put into production.
+
+Threads within Python
+~~~~~~~~~~~~~~~~~~~~~
+
+It is tempting to use threads within Python to perform background activity or to allow processing of data sets in parallel, however there is an issue with threading in Python, the Python Global Interpreter Lock or GIL. The GIL prevents two Python statements from being executed within the same interpreter by two threads simultaneously. Because we use a single interpreter for all Python code running in each service within Fledge, if a Python thread is created that performs CPU intensive work within it, we block all other Python code from running within that Fledge service.
```
Comment (pintomax) [inline docs/index.rst:248]: It seems something is missing between "however there is an issue with threading in Python, " and "the Python Global Interpreter Lock or GIL"

**Review by pintomax — CHANGES_REQUESTED** [review]

File: docs/index.rst [hunk 2]
```diff
@@ -155,14 +171,115 @@ Examining the content of the Python, a few things to note are;
...
+
+Warnings raised will also be logged to the error log but will not cause data to cease flowing through the pipeline.
+
+To view the error log you may examine the file directly on your host machine, for example */var/log/syslog* on a Ubuntu host, however it is also possible to view the error logs specific to Fledge from the Fledge user interface. Select the *System* option under *Logs* in the left hand menu pane. You may then filter the logs for a specific service to see only those logs that refer to the service which uses the filter you are interested in.
+
++-------------+
+| |logview_1| |
++-------------+
+
+Alternatively if you open the dialog for the service in the *South* or *North* menu items you will see two icons displayed in the bottom left corner of the dialogue that lets you alter the configuration of the service.
```
Comment (ashish-jabble) [inline docs/index.rst:270]: dialogue - > dialog

File: docs/index.rst [hunk 3]
```diff
@@ -9,11 +11,23 @@
...
 
-A general code syntax for the function that should be provided is;
+The functionality provided by this filters gives the user the opportunity to write Python code that can manipulate the reading data as it flows, however that modification should follow the same guidelines and principles as followed in the filters that come supplied as part of the Fledge distribution or are contributed by other Fledge users. The overriding principles however are
+
+  - Do not duplicate existing functionality provided by existing filters.
+
+  - Keep the operations small and focused. It is better to have multiple filters each with a specific purpose than to create large, complex Python scripts.
+
+  - Do not buffer large quantities of data, this will effect the footprint of the service and also slow the data pipeline.
```
Comment (ashish-jabble) [inline docs/index.rst:24]: Overriding guidelines we have already under below section Scripting Guidelines.
We can remove it from here

File: docs/index.rst (hunk 1 above)
Comment (MarkRiddoch) [reply docs/index.rst:248]: No, nothing is missing.

File: docs/index.rst (hunk 3 above)
Comment (MarkRiddoch) [reply docs/index.rst:24]: People will probably not read to the end, hence this is here to stress importance.

**Review by pintomax — APPROVED** [review]

**Review by ashish-jabble — APPROVED** [review]

**Pull request merged**